edgecast/waf/access_rule.go

```diff
@@ -1,199 +1,121 @@
-	DisallowedHeaders []string `json:"disallowed_headers,omitempty"`
-
-	// Contains access controls for IPv4 and/or IPv6 addresses.
-	// Specify each desired IP address using standard IPv4/IPv6 and CIDR notation.
-	IPAccessControls *AccessControls `json:"ip,omitempty"`
-
-	// MaxFileSize Indicates the maximum file size, in bytes, for a POST request body.
-	MaxFileSize int `json:"max_file_size,omitempty"`
-
-	// Assigns a name to this access rule.
-	Name string `json:"name,omitempty"`
-
-	// Contains access controls for referrers.
-	// All entries are regular expressions.
-	RefererAccessControls *AccessControls `json:"referer,omitempty"`
-
-	// Determines the name of the response header that will be included with blocked requests.
-	ResponseHeaderName string `json:"response_header_name,omitempty"`
-
-	// Contains access controls for URL paths.
-	// All entries are regular expressions.
-	URLAccessControls *AccessControls `json:"url,omitempty"`
-
-	// Contains access controls for user agents.
-	// All entries are regular expressions.
-	UserAgentAccessControls *AccessControls `json:"user_agent,omitempty"`
-}
-
-// AccessControls contains lists that identify traffic for access control
-type AccessControls struct {
-	// Contains entries that identify traffic that may access your content upon passing a threat assessment.
-	Accesslist []interface{} `json:"accesslist"`
-
-	// Contains entries that identify traffic that will be blocked or for which an alert will be generated.
-	Blacklist []interface{} `json:"blacklist"`
-
-	// Contains entries that identify traffic that may access your content without undergoing threat assessment.
-	Whitelist []interface{} `json:"whitelist"`
-}
+/*
+	This file contains operations and types specific to WAF Access Rules.
 
-// AddAccessRule creates an access rule that identifies valid or malicious requests via whitelists, accesslists, and blacklists.
-func (svc *WAFService) AddAccessRule(accessRule AccessRule) (*AddRuleResponse, error) {
-	url := fmt.Sprintf("/v2/mcc/customers/%s/waf/v1.0/acl", accessRule.CustomerID)
+	Access Rules (ACLs) identify valid or malicious requests via whitelists,
+	accesslists, and blacklists.
 
-	request, err := svc.Client.BuildRequest("POST", url, accessRule)
+	For detailed information about Access Rules in WAF, please refer to:
+	https://docs.edgecast.com/cdn/#Web-Security/Access-Rules.htm
+*/
 
-	if err != nil {
-		return nil, fmt.Errorf("AddAccessRule: %v", err)
-	}
-
-	parsedResponse := &AddRuleResponse{}
+import (
+	"fmt"
 
-	_, err = svc.Client.SendRequest(request, &parsedResponse)
+	"github.com/EdgeCast/ec-sdk-go/edgecast/client"
+)
 
+// AddAccessRule creates a new Access Rule for the provided account number
+// and returns the new rule's system-generated ID
+func (svc WAFService) AddAccessRule(
+	params AddAccessRuleParams,
+) (string, error) {
+	parsedResponse := &AccessRuleAddOK{}
+	_, err := svc.client.Do(client.DoParams{
```

Comment:
Something more meaningful?
svc.client.SubmitRequest(client.RequestParams{...})

Reply:
Hmm maybe... I used Do because that is the verb I commonly see in HTTP client libraries. I wouldn't mind changing it though.

Reply:
@changgyu-oh updated to SubmitRequest